### ISSUE TITLE
SQLFluff does not support Spark SQL struct field access

### ISSUE THREAD

**Issue by maoxingda** [sqlfluff/sqlfluff#6913]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
select
    1 as c1,
    'a' as c2,
    ELEMENT_AT(FROM_JSON('[{"f1":"v1","f2":"v2"}]', 'ARRAY<STRUCT<f1: STRING, f2: STRING>>'), -1).f1 as c3;
```

### Expected Behaviour

```
select
    1                                                                                                as c1,
    'a'                                                                                              as c2,
    ELEMENT_AT(FROM_JSON('[{"f1":"v1","f2":"v2"}]', 'ARRAY<STRUCT<f1: STRING, f2: STRING>>'), -1).f1 as c3;
```

### Observed Behaviour

```==== finding fixable violations ====
== [/Users/maoxd/code/redshift/surcharge/sql_files/dw_modeling_new/dwd/user_orders_iceberg.sql] FAIL
L:   2 | P:   6 | LT01 | 'alias_expression' elements are expected to be aligned.
                       | Found incorrect whitespace before 'as' keyword: ' '.
                       | [layout.spacing]
==== no fixable linting violations found ====
All Finished!
  [2 unfixable linting violations found]
```

### How to reproduce

sqlfluff fix ...

### Dialect

sparksql

### Version

```
sqlfluff version          
                  
3.3.1
```

### Configuration

```
[sqlfluff]
dialect = sparksql
max_line_length=128
large_file_skip_byte_limit = 0

[sqlfluff:indentation]
indented_ctes = True
indented_joins = False
indented_using_on = False

[sqlfluff:layout:type:alias_expression]
# We want non-default spacing _before_ the alias expressions.
spacing_before = align
# We want to align them within the next outer select clause.
# This means for example that alias expressions within the FROM
# or JOIN clause would _not_ be aligned with them.
align_within = select_clause
# The point at which to stop searching outward for siblings, which
# in this example would likely be the boundary of a CTE. Stopping
# when we hit brackets is usually a good rule of thumb for this
# configuration.
align_scope = bracketed
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
